### ISSUE TITLE
Roslyn doesn't warn about unsatisfied [MemberNotNull] annotations on autoproperties

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#49983]
**Version Used**: 
```
Branch master (8 Dec 2020)
Latest commit 60d947d by AlekseyTs:
Ensure that each generic local function gets its own container to cache dynamic call-sites. (#49802)

Fixes #49599.
```

**Steps to Reproduce**:

Compile and run the following code
```
#nullable enable
using System.Diagnostics.CodeAnalysis;

var d = new D();
d.Init = null;
d.F.ToString();

class C
{
  public string? F;

  [MemberNotNull(nameof(F))] public virtual string? Init { get => F = ""; set => F = ""; }
}

class D : C
{
  public override string? Init { get; set; }
}
``` 

**Expected Behavior**:
`warning CS8774: Member 'F' must have a non-null value when exiting.` reported for `D.Init`

**Actual Behavior**:
No warnings. The program crashes at runtime with a `NullReferenceException`

**Notes**
It might be very easy not to notice such annotations e.g. when overriding a property from a referenced assembly.
Interestingly I'm sure the warning used to be reported there in some of the previous versions of Roslyn

**Comment by RikkiGibson** [comment]
Related to #50244
